src/Dungeon/DungeonGenerator.elm

```diff
@@ -1,358 +1,573 @@
-                        , occupied = addActivePointToOccupiedTiles (ActiveCorridor corridor) model.occupied
-                    }
-            in
-            generateActivePointFromCorridor corridor model
-                |> Random.map
-                    (\activePoint ->
-                        case activePoint of
-                            (ActiveCorridor corridor) as activePoint ->
-                                if canFitCorridor modelWithoutActiveCorridor corridor then
-                                    { modelWithoutActiveCorridor
-                                        | activePoints = activePoint :: remainingPoints
-                                        , occupied = addActivePointToOccupiedTiles (ActiveCorridor corridor) modelWithoutActiveCorridor.occupied
-                                    }
-                                else
-                                    model
-
-                            (ActiveRoom room _) as activePoint ->
-                                if canFitRoom model room then
-                                    { modelWithInactiveCorridor
-                                        | activePoints = activePoint :: remainingPoints
-                                    }
-                                else
-                                    model
-                    )
-
-
-corridorStartFromRoomEntrance : Room -> Entrance -> ( Vector, Direction )
-corridorStartFromRoomEntrance room entrance =
-    let
-        roomEntrancePosition =
-            Entrance.position entrance
+{-| A corridor can have one of the three orientations:
+horizontal, vertical, diagonal (45 deg)
 
-        roomEntranceFacing =
-            Room.entranceFacing room entrance
+Therefore, between two points, the shortest path is either:
 
-        corridorEntranceFacing =
-            Vector.oppositeDirection roomEntranceFacing
+1.  A straight line if the x or y axis align or they both align in a 45 deg fashion.
+    ( eg ( 0, 0) to ( 5, 5) )
+2.  A bend with the bend either being 45deg or 90deg.
+    eg from (0,0) to (2, 3) there are 4 possible outcomes with one bend ending at (2, 3)
+    a. along the y (0, 1) and (0,3)
+    b. along the x (2, 0) and (2, 2)
 
-        corridorStartPosition =
-            roomEntranceFacing
-                |> Vector.fromDirection
-                |> Vector.add roomEntrancePosition
+An additional requirement is that the path leading out of the room cannot turn 90deg right away.
+
+-}
+moveFn : Dungeon -> PathData -> EverySet PathData
+moveFn dungeon ({ currentPosition, goal, validDirections, directionChanges } as pathData) =
+    let
+        --        _ =
+        --            Debug.log "Path: " currentPosition
+        nextPositions =
+            possiblePaths validDirections currentPosition goal
+                |> List.filter (flip Config.withinDungeonBounds dungeon.config)
+                |> List.filterMap (\movedTo -> toLastUnobstructedTile dungeon (Vector.path currentPosition movedTo))
```

Comment:
prolly better to do these as one `filterMap` if you can get away with it, because Elm doesn't do fusion, so this will do two "passes", I think, and this might speed things up quite a bit.